### PR TITLE
Sort members in the member list by power level and show mods and admins.

### PR DESCRIPTION
**Pull request by pixlwave**
First part of #2355, this PR makes 2 changes:
- Sort the members list by Power Level (and alphabetically as a secondary sort)
- Shows a label when the user's role is mod or admin.

Screenshot 2024-02-09 at 4 59 47 pm

### CODE REVIEW

File: ElementX/Sources/Services/Room/RoomMember/RoomMemberProxyProtocol.swift [hunk 1]
```diff
@@ -30,6 +30,8 @@ protocol RoomMemberProxyProtocol: AnyObject {
     var membership: MembershipState { get }
     var isAccountOwner: Bool { get }
     var isIgnored: Bool { get }
+    var powerLevel: Int { get }
+    var role: RoomMemberRole { get }
```
Comment (pixlwave) [inline ElementX/Sources/Services/Room/RoomMember/RoomMemberProxyProtocol.swift:34]: I was undecided if this should be the Rust type, or mapped to our own type. We seem to have other simple enums from Rust at this stage so I followed that.

File: ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift [hunk 2]
```diff
@@ -141,3 +142,22 @@ private struct RoomMembersDetails {
...
+    func sorted() -> Self {
+        sorted { lhs, rhs in
+            if lhs.powerLevel != rhs.powerLevel {
+                lhs.powerLevel > rhs.powerLevel
+            } else {
+                lhs.sortingName < rhs.sortingName
+            }
+        }
+    }
+}
```
Comment (Velin92) [inline ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift:156]: I would actually add this into the RoomMemberProxyProtocol extension directly.

File: ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift [hunk 3]
```diff
@@ -141,3 +142,22 @@ private struct RoomMembersDetails {
...
+
+private extension RoomMemberProxyProtocol {
+    var sortingName: String {
+        // If there isn't a displayname we sort by the userID without the @.
+        displayName ?? String(userID.dropFirst())
+    }
+}
```
Comment (Velin92) [inline ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift:163]: same

**Review by Velin92 — APPROVED** [review]
LGTM

File: ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift (hunk 2 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift:156]: Ok will do 👍 (I hesitated as wasn't sure if it was specific to this screen).

File: ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift (hunk 3 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/RoomMemberListScreen/RoomMembersListScreenViewModel.swift:163]: To me, `sortedName` would be an existing name that has been sorted. Whereas this is the name used for sorting.

**Pull request merged**